packages/engine/lib/memory/src/shared_memory/shared_memory.rs

```diff
@@ -54,7 +54,8 @@ impl<'id> MemoryId<'id> {
 
     /// Clean up generated shared memory segments associated with a given `MemoryId`.
     pub fn clean_up<Id: Borrow<Uuid>>(id: Id) -> Result<()> {
-        // TODO: Mac differences in shared_memory
+        // TODO: macOS does not store the shared memory FDs at `/dev/shm/`. Maybe it's not storing
+        //       FDs at all. Find out if they are stored somewhere and remove them instead.
```

Comment:
```suggestion
        //       FDs at all. Find out if they are stored somewhere and remove them instead,
        //       otherwise we have to figure out a way to remove them without relying on the
        //       file-system
```
I also don't think we need such intense indentation on the comments by the way

Reply:
I like to maintain an indentation of two spaces to keep syntax highlighting at JetBrains Product. Updated it accordingly.

Reply:
Yes but don't you have 7 spaces here?

Reply:
Github was showing me an old commit, apologies